eclipse-collections/src/main/java/org/eclipse/collections/impl/collector/Collectors2.java

```diff
@@ -784,6 +784,40 @@ private Collectors2()
                 EMPTY_CHARACTERISTICS);
     }
 
+    /**
+     * <p>Same as {@link #groupBy(Function, Supplier)}, except the result of evaluating groupBy function should return a
+     * unique key, or else an exception is thrown.</p>
+     *
+     * <p>Equivalent to using @{@link RichIterable#groupByUniqueKey(Function, MutableMap)}</p>
+     */
+    public static <T, K, R extends MutableMultimap<K, T>> Collector<T, ?, R> groupByUniqueKey(
```

Comment:
This is the correct return signature for groupByUniqueKey: `R extends MutableMap<K, T>`